JoinColony/colonyDapp
# Extension Manager Upgrade Extensions

This PR adds in the ability for the _Extension Manager_ to be able to upgrade currently installed extensions

In order to save time, this PR cherry-pick commits 0e3c3fffb1c12e7765730a4531ab8f9793269e7a 54cd6d98ecdf95d764dfbb74e4f50984bb55f643 and 4acd0d2b2a190a0bdf3e79d6544750cec7f08d76 from #2413

**Changes** 

- [x] use the new version `4` of `colonyJS`
- [x] add `ExtensionUpgrade` subcomponent
- [x] add `colonyExtensionUpgrade` saga
- [x] add resolver to fetch the latest extension version from the network
- [x] refactor extension to display correct version

**To test the extension version upgrade**

_(make sure to run `npm i` and `npm provision first)_

ColonyNetwork:
- After the stack was started
- Checkout `colonyNetwork` to commit `840e378c` _(I used a separate repo, but I guess the one in `src/lib/colonyNetwork` works fine as well...)_
- `yarn` _(to install packages)_
- run `yarn run truffle exec ./scripts/deployOldExtensionVersions.js`

Dapp:
- inside `src/modules/dashboard/sagas/extensions.ts`, change line `96` from:
```js
params: [getExtensionHash(extensionId), networkExtensionVersion],
```
to
```js
params: [getExtensionHash(extensionId), 1],
```
- install the _"Voting Reputation"_ extension _(either this or Coin Machine, they are the only ones that have version 2 available)_
- Enable it
- Upgrade it
- :tada: 

**Demo**

![Screenshot from 2021-03-13 21-20-28](https://user-images.githubusercontent.com/1193222/111041669-0914d400-8442-11eb-8c05-3741af0a04c6.png)

Resolves DEV-257
